web/modules/custom/joinup_core/js/sign_in_destination.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file
+ * Attaches a destination parameter to the sign in link to the current page.
+ */
+
+(function ($, Drupal, drupalSettings) {
+
+  /**
+   * Alters the sign in link to have a destination parameter to the source page.
+   *
+   * The home page 'Sign in' link and the 'Sign in' page 'Sign in' link are not
+   * altered.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attaches shared content behaviors.
+   */
+  Drupal.behaviors.signInDestination = {
+    attach: function (context) {
+      var path = drupalSettings.path;
+      // In these urls, only the query parameters will be copied if there is
+      // no destination parameter set.
```

Comment:
This is not clear actually. Actually, we copy all the query parameters if the destination parameter is not set.